include/zenoh_commons.h

```diff
@@ -667,6 +667,34 @@ typedef struct z_owned_scouting_config_t {
 typedef struct z_subscriber_t {
   const struct z_owned_subscriber_t *_0;
 } z_subscriber_t;
+/**
+ * The options for `zc_liveliness_declare_token`
+ */
+typedef struct zc_owned_liveliness_declaration_options_t {
+  uint8_t _inner;
+} zc_owned_liveliness_declaration_options_t;
```

Comment:
The `zenoh-commons.h` header is meant to be common across `zenoh-c` and `zenoh-pico`.
All functions starting with `zc_` are inherently meant to be specific to `zenoh-c`.
All these function declarations should hence go into [zenoh-concrete.h](https://github.com/eclipse-zenoh/zenoh-c/blob/master/include/zenoh_concrete.h).

@p-avital @milyin I see that other `zc_` functions slipped in `zenoh-commons.h` in the past. I think they should be moved into `zenoh-concrete.h` as well. Opinions?

Reply:
Yeah, the original goal was to be able to copy-paste this header into pico to ensure we always have the same signatures, but it didn't end up working out that way.

I think we can make the splitter logic a bit better if we want to stick to the original ideal. Otherwise, we could remove the splitter entirely, or just leave it in that state. Any odd these solutions is fine by me :)

Reply:
`zenoh-pico` has evolved quite a bit since we initially made that split and I see how the split could hardly work out in practice. Therefore I'm more in favour of unifying all the includes in a single file rather than introducing artificial splits (although correct in principle).

For what concerns the scope of this specific PR, let's not change anything for the time being... A different PR may follow in the future.